fix(splunk): treat HTTP 403 Forbidden as a permanent error

l00py opened the pull request:
#### Description
- Splunk responses with a 403 typically indicate an authentication or authorization issue that is not likely to be resolved by retrying. This change ensures that the error is treated as permanent to avoid unnecessary retries.
- This change is applied at `internal/splunk` level
- There should be no breaking changes for `signalfxexporter`, `splunkhecexporter`

#### Link to tracking issue
Fixes #39037 

#### Testing
Added and updated tests for handling 403 permanent error

#### Documentation
https://docs.splunk.com/Documentation/Splunk/latest/Data/TroubleshootHTTPEventCollector#Possible_error_codes

dehaansa reviewed (commented):
Looks good overall, but probably worth a Changelog entry even if it's _likely_ to be unnoticed by users.

atoulme commented:
Please add a changelog and take a look at the CI.

Commit:
fix(splunk): treat HTTP 403 Forbidden as a permanent error

Splunk responses with a 403 typically indicate an authentication or authorization issue that is not likely to be resolved by retrying. This change ensures that the error is treated as permanent to avoid unnecessary retries.

dehaansa reviewed (approved):
LGTM!
